sdk/keyvault/azkeys/models.go

```diff
@@ -397,3 +398,22 @@ type LifetimeActionsTrigger struct {
 	// Time before expiry to attempt to rotate or notify. It will be in ISO 8601 duration format. Example: 90 days : "P90D"
 	TimeBeforeExpiry *string `json:"timeBeforeExpiry,omitempty"`
 }
+
+func convertGeneratedMap(m map[string]*string) map[string]string {
+	ret := map[string]string{}
+
+	for k, v := range m {
+		ret[k] = *v
+	}
+
+	return ret
+}
+
+func convertToGeneratedMap(m map[string]string) map[string]*string {
+	ret := map[string]*string{}
+	for k, v := range m {
+		ret[k] = to.StringPtr(v)
+	}
+
+	return ret
+}
```

Comment:
Is this maybe something the code generator should do, or even just not use a `map[string]*string`? I'm a bit confused why we'd even define generate `Tags` as a map of string pointer values given I doubt any of them accept missing values. Testing against a resource group and a Key Vault, tags could have an empty value, but even via JSON that comes back as an empty string. Shouldn't that empty string be reflected in Go as well, or have you universally chosen to represent it as `nil`?

```json
  "tags": {
    "test": ""
  },
```

I would say: do whatever is idiomatic for Go, but in my experience, it's been more common to see empty strings than `nil` for strings.

/cc @JeffreyRichter

Reply:
There is a general problem that some maps (notably in ARM) use a `nil` value as the means to indicate "delete this key/value pair".  Unfortunately, not all specs are properly annotated with `x-nullable` to indicate this.  There's also the consideration of versioning tolerance if we were to support `x-nullable` in the future (i.e. should flipping this be a breaking change).

At any rate, we're looking at ways to improve this scenario as the common case has to pay the pointer-to-type tax which is unfortunate.